skbio/metadata/_intersection.pyx

```diff
@@ -0,0 +1,513 @@
+            self.minstart = imin3(self.start, self.cright.minstart, self.cleft.minstart)
+        elif self.cright is not EmptyNode:
+            self.maxend = imax2(self.end, self.cright.maxend)
+            self.minend = imin2(self.end, self.cright.minend)
+            self.minstart = imin2(self.start, self.cright.minstart)
+        elif self.cleft is not EmptyNode:
+            self.maxend = imax2(self.end, self.cleft.maxend)
+            self.minend = imin2(self.end, self.cleft.minend)
+            self.minstart = imin2(self.start, self.cleft.minstart)
+
+    def intersect( self, int start, int end, sort=True ):
+        """
+        given a start and a end, return a list of features
+        falling within that range
+        """
+        cdef list results = []
+        self._intersect( start, end, results )
+        return results
+
+    find = intersect
+
+    cdef void _intersect( IntervalNode self, int start, int end, list results):
+        # Left subtree
+        if self.cleft is not EmptyNode and self.cleft.maxend > start:
+            self.cleft._intersect( start, end, results )
+        # This interval
+        if ( self.end > start ) and ( self.start < end ):
+            results.append( self.interval )
+        # Right subtree
+        if self.cright is not EmptyNode and self.start < end:
+            self.cright._intersect( start, end, results )
+
+    cpdef void update(IntervalNode self, int start, int end,
+                      object old_feature, object new_feature):
+        """
+        given a start and end, replace all objects that
+        match the old_feature with new_feature.
+        """
+
+        # Left subtree
+        if self.cleft is not EmptyNode and self.cleft.maxend > start:
+            self.cleft.update( start, end, old_feature, new_feature )
+        # This interval
+        if ( self.end > start ) and ( self.start < end ):
+            if self.interval == old_feature:
+                self.interval = new_feature
+        # Right subtree
+        if self.cright is not EmptyNode and self.start < end:
+            self.cright.update( start, end, old_feature, new_feature )
+
+    cdef void _seek_left(IntervalNode self, int position, list results, int n, int max_dist):
+        # we know we can bail in these 2 cases.
+        if self.maxend + max_dist < position:
+            return
+        if self.minstart > position:
+            return
+
+        # the ordering of these 3 blocks makes it so the results are
+        # ordered nearest to farest from the query position
+        if self.cright is not EmptyNode:
+            self.cright._seek_left(position, results, n, max_dist)
+
+        if -1 < position - self.end < max_dist:
+            results.append(self.interval)
+
+        # TODO: can these conditionals be more stringent?
+        if self.cleft is not EmptyNode:
+                self.cleft._seek_left(position, results, n, max_dist)
```

Comment:
since this is absorbed code, are we not going to care about stylistic issues with it?